core/src/test/scala/pureconfig/data/instances/package.scala

```diff
@@ -0,0 +1,14 @@
+package pureconfig.data
+
+import pureconfig.ConfigConvert
+import pureconfig.error.{ CannotConvert, ConfigValueLocation }
+
+package object instances {
+
+  private val failConfigReadPercentage =
+    (s: String) => (l: Option[ConfigValueLocation]) =>
+      Left(CannotConvert(s, "Percentage", "Percentage is a dummy type, you should not read it", l, None))
+
+  implicit val percentageConfigWriter =
+    ConfigConvert.fromNonEmptyStringConvert[Percentage](failConfigReadPercentage, percentage => s"${percentage.value} %")
+}
```

Comment:
I suggest moving this `ConfigConvert` definition to a companion object of the `Percentage` class. It took me a while to figure out where you were redefining the default `case class` derivation logic for this to work with `checkArbitrary2`. Also, it removes the need for the explicit import in `BasicConvertersSuite`.